Refactor AsyncValidationCE module for bugfix

1eyewonder opened the pull request:
## Proposed Changes

Refactored the AsyncValidationCE source overloads to fix the issue seen in the last Amplifying F# video where AsyncResults were not binding correctly

## Types of changes

What types of changes does your code introduce to FsToolkit.ErrorHandling?
_Put an `x` in the boxes that apply and remove ones that don't apply_ 

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)


## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [X] Build and tests pass locally
- [X] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Further comments

⚠️ Please sanity check me. I think I updated a test case correctly, but because I am modifying a previous test case, I would appreciate a second set of eyes on this

TheAngryByrd commented on tests/FsToolkit.ErrorHandling.Tests/AsyncValidationCE.fs:
```diff
@@ -158,11 +158,12 @@ let ``AsyncValidationCE bind Tests`` =
             let innerData = "Foo"
-            let expected = Error [ innerData ]
+            let error = Error innerData
+            let expected = Error [ [ innerData ] ]
```
This doesn't seem right. It shouldn't be a double list.

1eyewonder commented:
@TheAngryByrd assuming build server works fine, I think this is good to look at again. I believe I fixed the issue mentioned above and noticed I didn't even include the task bindings initially. I went ahead and added those along with cleaning up some unnecessary `Result` namespace qualifiers. Let me know if you want anything done differently and sorry for such a long delay!

TheAngryByrd commented on tests/FsToolkit.ErrorHandling.Tests/AsyncValidationCE.fs:
```diff
@@ -474,19 +476,84 @@ let ``AsyncValidationCE applicative tests`` =
             Expect.equal actual (Ok 4) "Should be ok"
         }
 
+        testCaseAsync "Happy Path Async Result/Async Result"
```
Nit: These aren't really happy path if we're expecting errors 😄

TheAngryByrd commented on tests/FsToolkit.ErrorHandling.Tests/AsyncValidationCE.fs:
```diff
@@ -474,19 +476,84 @@ let ``AsyncValidationCE applicative tests`` =
+                asyncValidation {
+                    let! _ =
+                        async {
+                            do! Async.Sleep(1000)
+                            printfn "Hello"
```
```suggestion

```

TheAngryByrd commented on tests/FsToolkit.ErrorHandling.Tests/AsyncValidationCE.fs:
```diff
@@ -474,19 +476,84 @@ let ``AsyncValidationCE applicative tests`` =
+
+                    and! _ =
+                        async {
+                            do! Async.Sleep(1000)
+                            printfn "World"
```
```suggestion

```

TheAngryByrd commented on tests/FsToolkit.ErrorHandling.Tests/AsyncValidationCE.fs:
```diff
@@ -474,19 +476,84 @@ let ``AsyncValidationCE applicative tests`` =
+                asyncValidation {
+                    let! _ =
+                        task {
+                            do! Async.Sleep(1000)
+                            printfn "Hello"
```
```suggestion

```

TheAngryByrd commented on tests/FsToolkit.ErrorHandling.Tests/AsyncValidationCE.fs:
```diff
@@ -474,19 +476,84 @@ let ``AsyncValidationCE applicative tests`` =
+
+                    and! _ =
+                        task {
+                            do! Async.Sleep(1000)
+                            printfn "World"
```
```suggestion
```

TheAngryByrd reviewed (changes requested):
Thanks for fixing it up! Left some feedback.
